### PR TITLE
Add suffix-free domain separation for H_1.

### PR DESCRIPTION
**Pull request by chris-wood**
See https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-hash-to-curve-08#section-10.3.

### CODE REVIEW

File: poc/oprf.sage [hunk 1]
```diff
@@ -164,12 +164,14 @@ class Client(object):
...
 		encoded_point = self.suite.group.serialize(y)
 
-		h.update(I2OSP(len(finalize_dst), 2))
-		h.update(finalize_dst)
 		h.update(I2OSP(len(x), 2))
 		h.update(x)
 		h.update(I2OSP(len(encoded_point), 2))
 		h.update(encoded_point)
+		h.update(I2OSP(len(aux), 2))
+		h.update(aux)
```
Comment (alxdavids) [inline poc/oprf.sage:172]: shall we rename this to aux as well?

Comment (chris-wood) [reply poc/oprf.sage:172]: Hmm? It is already named aux.

**Comment by armfazh** [comment]
Does the purpose of the `info` variable must be user-specific, rather than use it as domain separation tag? 
One can define a separation tag without requiring the user to care about it. do you think this could work?

File: poc/oprf.sage (hunk 1 above)
Comment (alxdavids) [reply poc/oprf.sage:172]: oops, I meant `info`.

Comment (chris-wood) [reply poc/oprf.sage]: Doh, yes, good catch! Fixed.

**Comment by chris-wood** [comment]
> Does the purpose of the info variable must be user-specific, rather than use it as domain separation tag?

`info` is an application-specific domain separation string. If absent, we still require domain separation, so there's "two" layers of separation in play here.

**Comment by chris-wood** [comment]
@alxdavids if you're OK with these changes, please feel free to merge!

**Pull request merged**